src/System.Memory/tests/ArrayBufferWriter/ArrayBufferWriterTests.Char.cs

```diff
@@ -0,0 +1,303 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Buffers.Tests
+{
+    public static partial class ArrayBufferWriterTests_Char
```

Comment:
Why are we duplicating all of these tests for different data types?  Wouldn't it be better to just have a generic base test type parameterized on T, and then have multiple derived types that provide T==char, T==string, etc.?  This is a ton of what appears to be largely duplicated code.

Reply:
Yep, definitely. The only difference is that I have stream-based tests for T=byte, but that can be extracted out.